LibSerialize.lua

```diff
@@ -836,20 +842,8 @@ end
 function LibSerializeInt:_ReadObject()
     local value = self:_ReadByte()
 
-    if self._async then
-        if self._currentObjectCount then
-            self._currentObjectCount = self._currentObjectCount + 1
-        end
-        local elapsedTime
-        if self._currentElapsedTime and self._opts.timeFn then
-            elapsedTime = self._opts.timeFn()
-        end
-        if (elapsedTime and elapsedTime - self._currentElapsedTime > self._opts.yieldOnElapsedTime) or
-        (self._currentObjectCount and self._currentObjectCount > self._opts.yieldOnObjectCount) then
-            if elapsedTime then self._currentElapsedTime = elapsedTime end
-            if self._currentObjectCount then self._currentObjectCount = 0 end
-            coroutine_yield()
-        end
+    if self._async and self:_yieldCheckFn() then
```

Comment:
Allocate a separate "working table" table on `self` to pass in as an argument here - we don't want to leak the library's internals. So this would be something like `self._yieldCheckFn(self._asyncWorkingTable)`. It's not quite as convenient, but the alternative is someone inadvertently breaking the library by stepping on internal tracking state...